src/base/bittorrent/session.cpp

```diff
@@ -195,6 +194,35 @@ namespace
 
     template <typename T>
     LowerLimited<T> lowerLimited(T limit, T ret) { return LowerLimited<T>(limit, ret); }
+
+#if LIBTORRENT_VERSION_NUM >= 10100
+    std::string makeFingerprint(const char* peerId, int major, int minor, int revision, int tag)
+    {
+        Q_ASSERT(peerId);
+        Q_ASSERT(major >= 0);
+        Q_ASSERT(minor >= 0);
+        Q_ASSERT(revision >= 0);
+        Q_ASSERT(tag >= 0);
+        Q_ASSERT(std::strlen(peerId) == 2);
+
+        auto versionToChar = [](int v) -> char
+        {
+            if (v >= 0 && v < 10) return static_cast<char>('0' + v);
+            if (v >= 10) return static_cast<char>('A' + (v - 10));
+            Q_ASSERT(false);
+            return '0';
+        };
+
+        char s[100];
+        snprintf(s, 100, "-%c%c%c%c%c%c-",
```

Comment:
I have troubles counting '%c's in the format string and making sure their number is equal to the number of arguments (which will be checked by compiler though). As to me, the following is more readable:
```C++
std::ostring_stream buf;
buf << '-' 
    << peerId[0] << peerId[1]
    << versionToChar(major) << versionToChar(minor) << versionToChar(revision) << versionToChar(tag)
    << '-';
return buf.str();
```